consensus/istanbul/validator/weighted_random_test.go

```diff
@@ -128,6 +129,35 @@ func makeTestWeightedCouncil(weights []uint64) (valSet *weightedCouncil) {
 	return
 }
 
+func TestWeightedValidator_Copy(t *testing.T) {
+	testCases := []struct {
+		addr        common.Address
+		reward      common.Address
+		votingpower uint64
+		weight      uint64
+	}{
+		{
+			common.Address{},
+			common.Address{},
+			1000,
+			0,
+		},
+		{
+			common.StringToAddress("address"),
+			common.StringToAddress("reward address"),
+			1000,
+			10,
+		},
+	}
+	for _, testCase := range testCases {
+		validator := newWeightedValidator(testCase.addr, testCase.reward, testCase.votingpower, testCase.weight)
+		copiedValidator := validator.Copy()
+
+		assert.Equal(t, validator, copiedValidator)
+		assert.NotEqual(t, fmt.Sprintf("%p", &validator), fmt.Sprintf("%p", &copiedValidator))
```

Comment:
assert automatically checks the object which the pointer linked.
but what i want is checking if the addresses of variables are different

Reply:
reflected thanks!!